File: 프로그래머스/0/120903. 배열의 유사도/배열의 유사도.py
def solution(s1, s2):
    answer = 0  # 공통 원소 개수를 저장할 변수

    for i in s1:  # 첫 번째 리스트의 원소를 순회
        if i in s2:  # 두 번째 리스트에 동일한 원소가 존재하면
            answer += 1  # 공통 원소 개수 증가

    return answer  # 공통 원소 개수 반환

# 테스트 예제 실행
print(solution(["a", "b", "c"], ["com", "b", "d", "p", "c"]))   # 2
print(solution(["n", "omg"], ["m", "dot"]))                     # 0

File: 프로그래머스/0/120903. 배열의 유사도/배열의 유사도.java
public class Solution {

    public static int solution(String[] s1, String[] s2) {
        int answer = 0;

        for (int i = 0; i < s1.length; i++) {
            for (int j = 0; j < s2.length; j++) {
                if (s1[i].equals(s2[j])) {
                    answer++;
                    break; // 중복 방지를 위해 바로 탈출
                }
            }
        }

        return answer;
    }

    public static void main(String[] args) {
        System.out.println(solution(
            new String[]{"a", "b", "c"},
            new String[]{"com", "b", "d", "p", "c"}
        )); // 2

        System.out.println(solution(
            new String[]{"n", "omg"},
            new String[]{"m", "dot"}
        )); // 0
    }
}